### PR TITLE
Fix some issues

### PR DESCRIPTION
**Pull request by demkada**
Fix for issue #344 	948592a
Using the right AccessControlException class to throw Exception 	db10603

### CODE REVIEW

File: src/test/java/org/owasp/esapi/reference/ValidatorTest.java [hunk 1]
```diff
@@ -297,7 +297,7 @@ public void testGetValidSafeHTML() throws Exception {
 
         assertEquals("Test. &lt;<div>load=alert()</div>", rule.getSafe("test", "Test. <<div on<script></script>load=alert()"));
         assertEquals("Test. <div>b</div>", rule.getSafe("test", "Test. <div style={xss:expression(xss)}>b</div>"));
-        assertEquals("Test.", rule.getSafe("test", "Test. <s%00cript>alert(document.cookie)</script>"));
+        assertEquals("Test.", rule.getSafe("test", "Test. <script>alert(document.cookie)</script>"));
```
Comment (chrisisbeef) [inline src/test/java/org/owasp/esapi/reference/ValidatorTest.java]: Why did you drop the null-byte test here? Like the tests under it, this is verifying that the validation works even when a special character interrupts the payload. The null-byte attack is a valid payload to verify.

Comment (demkada) [inline src/test/java/org/owasp/esapi/reference/ValidatorTest.java]: It was a mistake. I wanted to change that line with 
assertEquals("Test. alert(document.cookie)", rule.getSafe("test", "Test. <s%00cript>alert(document.cookie)</script>")); for the test coherency to fix issue #344

Comment (demkada) [inline src/test/java/org/owasp/esapi/reference/ValidatorTest.java]: Hello,
Sorry for this absurdity. 
Is it normal that the assertEquals expected object does not correspond to the actual safe object return by the HTMLValidationRule class?

I think the expected object should be equal to "Test. alert(document.cookie)". What is your opinion? maybe I do not dig where I should.

**Comment by kwwall** [comment]
Closed via merge.

**Pull request merged**